File: server/dotnet/sdk-dotnet-example/Configuration/DIConfiguration.cs
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Options;
using OnlinePayments.Sdk;
using OnlinePayments.Sdk.Merchant;

namespace Configuration;

public static class DIConfiguration
{
    public static IServiceCollection AddConfiguration(this IServiceCollection services)
    {
        services.AddSingleton<IClient>(sp =>
        {
            AppSettings.AppSettings appSettings = sp.GetRequiredService<IOptions<AppSettings.AppSettings>>().Value;
            CommunicatorConfiguration communicatorConfiguration = new()
            {
                ApiEndpoint = new Uri(appSettings.ApiEndpoint),
                ApiKeyId = appSettings.ApiKey,
                SecretApiKey = appSettings.ApiSecret,
                Integrator = appSettings.MerchantId
            };

            return Factory.CreateClient(communicatorConfiguration);
        });

        services.AddSingleton<MerchantClientConfiguration.MerchantClientConfiguration>();
        services.AddSingleton<IMerchantClient>(sp =>
        {
            MerchantClientConfiguration.MerchantClientConfiguration merchantClientConfiguration =
                sp.GetRequiredService<MerchantClientConfiguration.MerchantClientConfiguration>();

            return merchantClientConfiguration.GetMerchantClient();
        });

        return services;
    }
}

File: server/php/src/Configuration/DIConfiguration.php
<?php

namespace OnlinePayments\ExampleApp\Configuration;

use DI\ContainerBuilder;
use DI;
use OnlinePayments\ExampleApp\Configuration\MerchantClientConfiguration\MerchantClientConfiguration;
use OnlinePayments\Sdk\Merchant\MerchantClient;

class DIConfiguration
{
    private MerchantClientConfiguration $merchantConfig;

    public function __construct(string $projectRoot)
    {
        $this->merchantConfig = new MerchantClientConfiguration($projectRoot);
    }

    public function configure(ContainerBuilder $builder): void
    {
        $builder->addDefinitions([
            MerchantClientConfiguration::class => DI\value($this->merchantConfig),

            MerchantClient::class => function() {
                return $this->merchantConfig->getMerchantClient();
            }
        ]);
    }
}
